onnx/onnx.proto

```diff
@@ -124,7 +139,17 @@ message GraphProto {
 
   // A human-readable documentation for this graph. Markdown is allowed.
   optional string doc_string = 10;
-};
+
+  // Argument information for input/output of this graph.
+  // Each ArgInfoProto corresponds one-to-one by position with the 
```

Comment:
If we are willing to do a BC breaking change, it would be better to bundle information about an input (or output) together: name, type, shape, and even default-values (which are currently in the initializer list), instead of correlating across lists by position or name etc. (with the possibility of missing or mismatching entries).

Reply:
I do have a strong preference to at least put the arginfo and name stuff together.

I'm less convicted on putting initializer in there given the other threads going on. 

Ed?